src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -86,11 +89,60 @@ public string GetActiveRepoPath()
             return repo?.Info?.Path ?? string.Empty;
         }
 
+        public IEnumerable<ISimpleRepositoryModel> GetKnownRepositories()
+        {
+            return PokeTheRegistryForRepositoryList();
+        }
+
+        const string TEGitKey = @"Software\Microsoft\VisualStudio\14.0\TeamFoundation\GitSourceControl";
+        static RegistryKey OpenGitKey(string path)
+        {
+            return Registry.CurrentUser.OpenSubKey(TEGitKey + "\\" + path, true);
+        }
+
+        static IEnumerable<ISimpleRepositoryModel> PokeTheRegistryForRepositoryList()
+        {
+            using (var key = OpenGitKey("Repositories"))
+            {
+                return key.GetSubKeyNames().Select(x =>
+                {
+                    using (var subkey = key.OpenSubKey(x))
```

Comment:
Isn't it possible that `subkey` could be null here? Or that `OpenGitKey` could throw an exception such as a `SecurityException`? We should make sure we're handling exceptions in the calls to `PokeTheRegistryForRepositoryList`